ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/shared/service/internal/PermissionsServiceImpl.java

```diff
@@ -45,6 +66,23 @@ public PermissionsServiceImpl(AuthenticationService authenticationService,
 
   @Override
   public String getFilePermissionsEntity(String stableId) {
+    
+    if (authenticationService.getAuthentication() instanceof OAuth2Authentication) {
+        OAuth2Authentication authentication = (OAuth2Authentication) authenticationService.getAuthentication();
+        if (authentication != null && authentication.getUserAuthentication() instanceof CustomUsernamePasswordAuthenticationToken) {
+            CustomUsernamePasswordAuthenticationToken athenticationToken = (CustomUsernamePasswordAuthenticationToken) authentication.getUserAuthentication();
+            Map<String, List<String>> datasetFileMapping = athenticationToken.getDatasetFileMapping();
+            if (datasetFileMapping != null) {
+                String datasetId = datasetFileMapping.entrySet().parallelStream()
+                        .filter(e -> e.getValue().contains(stableId)).map(Map.Entry::getKey).findFirst()
+                        .orElse(null);
+                if (datasetId != null) {
```

Comment:
Does it make sense to continue the execution of this function instead of returning a PermissionDeniedException id we don't find the dataset?

Reply:
There are 2 scenarios
1. The datasetId and fileId mapping get from AAI Token Introspection result
2. The datasetId and fileId mapping get from the database only.
 
The code has to work for both the scenarios we will not always get the mapping from AAI. So here if we don't find the dataset in AAI it means it could be the 2nd scenario so we hit the filedatabase service to get that mapping.